crates/bevy_ecs/src/system/commands.rs

```diff
@@ -281,22 +362,26 @@ impl Commands {
         self
     }
 
+    /// Adds a command directly to the command list. If `command` is boxed, call `add_command_boxed`.
```

Comment:
the difference between `add_command` and `add_command_boxed` is static and dynamic dispatch, the fact that one is boxed is a side effect of that...
see https://doc.rust-lang.org/1.8.0/book/trait-objects.html for more info on how that work.

the idea is that if you know the type of command at compile time, you can use `add_command`, but if your code can send any command type then you need `add_command_boxed`. maybe it could be renamed `add_command_dynamic` or something? not sure if that's clearer...

Reply:
Ah yes, I wasn't even thinking of static vs dynamic!

Should the existing usages of `self.commands.push(Box::new...`  in `Commands` be updated to use `self.add_command(...`?

Reply:
that falls into the realm of compile time optimisation, by calling `self.add_command` you add one level of indirection to compile away (inline) or not, plus you introduce static dispatch that increase compile time. I don't have a strong opinion of that and I'm not a fan of optimisation enough to measure it, but I think @cart can weight in

Reply:
I don't have a strong opinion, just figured it'd be worth following our own advice we're adding in the PR :grin: